app/components/works/edit/content_hierarchy_component.rb

```diff
@@ -4,15 +4,12 @@ module Works
   module Edit
     # Component for rendering an editable file hierarchy table for a Content.
     class ContentHierarchyComponent < ApplicationComponent
-      # NOTE: content is a reserved word in a view component, so using content_obj instead.
-      def initialize(content_obj:)
-        @content_obj = content_obj
+      def initialize(content_files:)
+        @content_files = content_files
         super()
       end
 
-      def content_files
-        @content_obj.content_files.path_order
-      end
+      attr_reader :content_files
```

Comment:
we need to pass in the content_files specifically now, because it may have a where search filter applied to the results ... we didn't need the full content object anyway for this view